src/commands/edge_app.rs

```diff
@@ -185,6 +185,37 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    pub fn promote(&self, manifest: &EdgeAppManifest, version: &Option<i32>) -> Result<i32, CommandError> {
+        let payload = match version {
+            Some(version_value) => json!(
+                {
+                    "revision": version_value,
+                    "app_id": manifest.app_id.clone(),
+                }
+            ),
+            None => json!(
+                {
+                    "app_id": manifest.app_id.clone(),
+                }
+            ),
+        };
+        let response = commands::post(
+            &self.authentication,
+            "v4/edge-apps/promote",
+            &payload,
+        )?;
+
+        if let Some(dict) = response.as_object() {
```

Comment:
you can reduce nesting here by transforming it in something like
let revision_value = response
    .as_object()
    .and_then(|dict| dict.get("updated"))
    .and_then(|revision| revision.as_i64())
    .map(|revision_value| revision_value as i32);